onedal/covariance/incremental_covariance.py

```diff
@@ -0,0 +1,61 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+import numpy as np
+
+from daal4py.sklearn._utils import daal_check_version, get_dtype, make2d
+from onedal import _backend
+
+from ..datatypes import _convert_to_supported, from_table, to_table
+from .covariance import BaseEmpiricalCovariance
+
+
+class IncrementalEmpiricalCovariance(BaseEmpiricalCovariance):
+    """
+    Incremental Covariance oneDAL implementation
+    """
+
+    def __init__(self, method="dense", bias=False):
+        super().__init__(method, bias)
+        self._partial_result = _backend.covariance.partial_compute_result()
+
+    def partial_fit(self, X, queue=None):
+        if not hasattr(self, "_policy"):
+            self._policy = self._get_policy(queue, X)
```

Comment:
Make sense on refactoring update _get_policy method.
Could be done as a follow up work